libs/unity/library/Runtime/Scripts/Media/AudioReceiver.cs

```diff
@@ -308,6 +308,10 @@ void IMediaReceiverInternal.DetachFromTransceiver(Transceiver transceiver)
         {
             Debug.Assert((_transceiver == null) || (_transceiver == transceiver));
             _transceiver = null;
+            _track = null;
```

Comment:
This change and the one in `VideoReceiver.cs` seem quite different, why are things so asymmetrical?

Reply:
VideoReceiver doesn't seem to _do_ anything except exposing the `Track` callback, so there isn't much that it can do on Enable/Disable, and the logic around it is absent.

This raises some interesting questions:
- should the Transceiver direction be controlled by the enabled state of the connected receiver? At the moment it isn't, but it's unfortunate that we keep streaming in a track that we aren't rendering.
- can we improve this? At the moment `AudioReceiver` merges the feature of a "receiver" (plugs into a MediaLine and forwards its frames) and a "renderer" (plays the audio), while for video this is done by two separate classes. So we could make things more coherent by:
  - splitting `AudioReceiver` into two classes like for video; clear but not great (we need two different components for audio too, we have a long chain of classes already)
  - merging `VideoRenderer` and `VideoReceiver` -  have an `AudioRenderer` (rename from `AudioReceiver`) which renders audio automatically and a `VideoRenderer` that renders video automatically and all direct access to frames can done in code (which is already the case) by accessing the remote track in the MediaLine. Sounds nice - allows us to fix the hierarchy quirks around VideoRendererSource - but there are a few issues: how do we attach two renderers to the same track? what UI we make to tell "I want to receive this track but don't have a Renderer object to attach"?
  - make `XReceiver` just expose the callbacks and have `XRenderer` inherit from it - then you plug a base receiver into the MediaLine if you only want the callbacks, or a renderer if you want the stream to be rendered automatically (or you write your own renderer class). Still difficult to attach multiple renderers though.

Thoughts?

Reply:
This has been refactored in a different PR.